api/integrations/slack/views.py

```diff
@@ -36,6 +37,7 @@
 class SlackGetChannelsViewSet(GenericViewSet):
     serializer_class = SlackChannelListSerializer
     pagination_class = None  # set here to ensure documentation is correct
+    permission_classes = [IsAuthenticated, NestedEnvironmentPermissions]
```

Comment:
According to default permissions set in `NestedEnvironmentPermissions`, anybody who can view the environment will be able to list Slack channels using credentials stored with Slack integration settings. Is this the desired behaviour or the permissions could be narrowed further to only allow this for project admins?

Reply:
Yeah, I think this is a fair suggestion. I did consider this, but the functionality didn't seem to exist 'off the shelf' in the codebase (which I was surprised by). I'll add it here though.